nixos/modules/services/networking/bind.nix

```diff
@@ -274,12 +274,9 @@ in
         ${pkgs.coreutils}/bin/mkdir -p ${cfg.directory}
         chown ${bindUser} ${cfg.directory}
       '';
-
-      serviceConfig = {
-        ExecStart = "${bindPkg.out}/sbin/named -u ${bindUser} ${optionalString cfg.ipv4Only "-4"} -c ${cfg.configFile} -f";
-        ExecReload = "${bindPkg.out}/sbin/rndc -k '/etc/bind/rndc.key' reload";
-        ExecStop = "${bindPkg.out}/sbin/rndc -k '/etc/bind/rndc.key' stop";
-      };
+      script = "${bindPkg.out}/sbin/named -u ${bindUser} ${optionalString cfg.ipv4Only "-4"} -c ${cfg.configFile} -f";
```

Comment:
Be aware that, because you are calling `named` with the `-f` option, it means, that this script will run in the foreground. If you append something to this script, it will essentially wait until `named` stops, for example because of a SIGTERM signal and then run the appended lines. I would say, that that can cause some confusion.

In sysmted there is a notioni for prepending scripts with `ExecStartPre` . Could this help with your use-case?

https://man7.org/linux/man-pages/man5/systemd.service.5.html

Reply:
I don’t really need the `script` option to allow prepending/appending, I mostly changed that to be consistent with the other two. The main option I want changed is `ExecStop` to `preStop` so I can prepend a “thaw” before bind exits. Currently if I try to do this I have to use `mkForce` and copy the existing stop script into my config: https://git.averywinters.org/avery/home/src/commit/582efb553711807a7ec99eafaf63000114688810/amsterdam/configuration.nix#L283

Reply:
And looking at how these new options get translated, it looked to me like `script` populates `ExecStart`, `reload` for `ExecReload` and `preStop` for `ExecStop`.

https://github.com/NixOS/nixpkgs/blob/2dc5c6ef045ad396ca0e6cce0258108541c78d5b/nixos/lib/systemd-unit-options.nix#L404